File: 新东方CCCJunior/lecture9/ReturningHome.java
import java.util.LinkedList;
import java.util.Scanner;

public class ReturningHome {

	public static void main(String[] args)
	{
		LinkedList<String> path = new LinkedList<String>();
		Scanner sc = new Scanner(System.in);
		
		String next;
		
		do{
			next = sc.nextLine();
			path.addFirst(next);
		}while(!next.equals("SCHOOL"));
		
		path.stream().forEach(str->System.out.println(str));
		
		for(int i = 1; i < path.size()-1; i+=2) {
			if(path.get(i).equals("L")) {
				System.out.println("Turn RIGHT onto " + path.get(i+1) + " Street.");
			}else {
				System.out.println("Turn LEFT onto " + path.get(i+1) + " Street.");
			}
		}
		
		if(path.get(path.size()-1).equals("L")) {
			System.out.println("Turn RIGHT into your HOME.");
		}else {
			System.out.println("Turn LEFT into your HOME.");
		}	
	}
}

File: prev_exams_and_ans/2005ccc/q3_returning_home/returning_home.py
stack = ["HOME"]

while True:
    instruction = input()
    if instruction == "SCHOOL":
        break
    elif instruction == "R":
        stack.append("LEFT")
    elif instruction == "L":
        stack.append("RIGHT")
    else:
        stack.append(instruction)


while len(stack) > 0:
    ins = stack.pop()
    loc = stack.pop()
    if (loc != "HOME"):
        print(f"Turn {ins} onto {loc} street.")
    else:
        print(f"Turn {ins} into your {loc}.")